control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -1149,9 +1148,15 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
         }
     }
 
-    async fn shutdown_nexus(&self, _: &ShutdownNexus) -> Result<(), SvcError> {
-        // todo: add implementation
-        Ok(())
+    async fn shutdown_nexus(&self, request: &ShutdownNexus) -> Result<(), SvcError> {
+        let dataplane = self.grpc_client_locked(request.id()).await?;
+        let result = dataplane.shutdown_nexus(request).await;
+        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+        self.update_nexus_states(ctx.deref_mut()).await?;
+        match result {
+            Ok(_) => Ok(()),
+            Err(error) => Err(error),
```

Comment:
need to consider idempotence here, what happens if the nexus is already shutdown; for now we could check if the nexus state is shutdown but that doesn't cover the case when shutdown is in progress but not complete.
@mtzaurus

Reply:
As discussed, We will have a separate PR to handle idempotency for this. We need to decide idempotency scenarios.